crates/router/src/core/payments/helpers.rs

```diff
@@ -1370,6 +1370,22 @@ fn validate_options_for_inequality<T: PartialEq>(
     )
 }
 
+pub fn validate_max_amount(
+    amount: api_models::payments::Amount,
+) -> CustomResult<(), errors::ApiErrorResponse> {
+    match amount {
+        api_models::payments::Amount::Value(value) => {
+            //max_amount allowed is 999999999 in minor units
+            utils::when(value.get() > 999999999, || {
```

Comment:
can you move this to a constant, we can make use of the same const value in all other places as well